File: app/src/main/java/dev/ragnarok/fenrir/model/selection/FileManagerSelectableSource.java
package dev.ragnarok.fenrir.model.selection;

import android.os.Parcel;


public class FileManagerSelectableSource extends AbsSelectableSource {

    public static final Creator<FileManagerSelectableSource> CREATOR = new Creator<FileManagerSelectableSource>() {
        @Override
        public FileManagerSelectableSource createFromParcel(Parcel in) {
            return new FileManagerSelectableSource(in);
        }

        @Override
        public FileManagerSelectableSource[] newArray(int size) {
            return new FileManagerSelectableSource[size];
        }
    };

    public FileManagerSelectableSource() {
        super(Types.FILES);
    }

    protected FileManagerSelectableSource(Parcel in) {
        super(in);
    }

    @Override
    public int describeContents() {
        return 0;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/model/selection/FileManagerSelectableSource.kt
package dev.ragnarok.fenrir.model.selection

import android.os.Parcel
import android.os.Parcelable

class FileManagerSelectableSource : AbsSelectableSource {
    constructor() : super(Types.FILES)
    internal constructor(parcel: Parcel) : super(parcel)

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<FileManagerSelectableSource> {
        override fun createFromParcel(parcel: Parcel): FileManagerSelectableSource {
            return FileManagerSelectableSource(parcel)
        }

        override fun newArray(size: Int): Array<FileManagerSelectableSource?> {
            return arrayOfNulls(size)
        }
    }
}